src/sign-in-blazorserver/README.md

```diff
@@ -0,0 +1,114 @@
+
+In _appsettings.json_, update each variable with values from the app registration you created earlier.
+
+```json
+"TenantId": "[Enter 'common', or 'organizations' or the Tenant ID (Obtained from the Azure portal. Select 'Endpoints' from the 'App registrations' blade and use the GUID in any of the URLs), e.g. da41245a5-11b3-996c-00a8-4d99re19f292]",
+"ClientId": "[Enter the Client Id (Application ID obtained from the Azure portal), e.g. ba74781c2-53c2-442a-97c2-3d60re42f403]",
+"ClientSecret": "[Copy the client secret added to the app from the Azure portal]",
+```
+
+## Run the application
+
+### 1. Run the webapp
+
+1. Execute the following command to get the app up and running:
+
+   ```bash
+   dotnet run
+   ```
+
+### 2. Signin into the Blazor Server application
+
+1. Once the web app is listening, navigate to https://localhost:5001. Click the Sign In button and complete the login process.
+
+   ![A screenshot of a Blazor Server application application guiding the user to click the "Sign In" button.](./app-signed-in.png)
+
+1. Click Sign Out
+
+   ![A screenshot of a Blazor Server application application guiding the user to click the "Sign Out" button.](./app-signed-out.png)
+
+
+## About the code
+
+This ASP.NET Core 6 Blazor Server application is created using .NET Blazor Server App template. The app is adding sign-in to protect itself, and as a consequence this is requiring the user to be authenticated in Azure Active Directory.
+
+When this .NET Blazor Server starts and before listening to any http call, it bootstraps the application using a single-surface API provided as part of the **Microsoft Identity Web** and the **Microsoft.Idenitty.Web.UI**. The former is tying ASP.NET Core, its authentication middleware for sign-in, and the [Microsoft Authentication Library (MSAL) for .NET](https://github.com/azuread/microsoft-authentication-library-for-dotnet), while the latter adds the UI components and controllers for user sing-in (enter credentials, update password, multi-factor auth, additional consents, etc) and sign-out. All the details required for authentication are being gathered from a configuration section named `AzureAd` as well as others entries used during the call to a protected API like the scopes. As of authorization, it is using the default policy options. Additionally, special services are injected [specifically for Blazor for re-signing, consent and conditional access purposes](https://github.com/AzureAD/microsoft-identity-web/wiki/Managing-incremental-consent-and-conditional-access#in-blazor-server). The recommended pattern to acquire tokens is firstly attempting to acquire them [non-interactevelly (or silent from cache), and then interactively](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-authentication-flows#interactive-and-non-interactive-authentication). Therefore, it is required to store the tokens for them to be cached. In this tutorial tokens are being stored in memory.
```

Comment:
```suggestion
When this .NET Blazor Server starts and before listening for any HTTP requests, it bootstraps the application using a single-surface API provided as part of **Microsoft.Identity.Web** and **Microsoft.Idenitty.Web.UI**. The former is sign-in middleware for ASP.NET Core that uses [Microsoft Authentication Library (MSAL) for .NET](https://github.com/azuread/microsoft-authentication-library-for-dotnet), while the latter adds UI components and controllers to facilitate user sign-in, sign-out, and other account experiences. All the details required for authentication are being gathered from a configuration section named `AzureAd` as well as others entries used during the call to a protected API, like the scopes. As for authorization, it is using the default policy options. Additionally, special services are injected [specifically for Blazor for re-signing, consent and conditional access purposes](https://github.com/AzureAD/microsoft-identity-web/wiki/Managing-incremental-consent-and-conditional-access#in-blazor-server). The recommended pattern to acquire tokens is first attempting to acquire them [non-interactevelly (or silent from cache), and then interactively](https://docs.microsoft.com/azure/active-directory/develop/msal-authentication-flows#interactive-and-non-interactive-authentication). Therefore, it is required to store the tokens for them to be cached. In this tutorial tokens are being stored in memory.
```

Reply:
I'm incorporating this, just leaving out a single suggested sentence. Please see [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-web-app-call-api-overview#specifics) and [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/microsoft-identity-web) for more info.  

After that, If you still consider your sentence was accurate, Im happy to incorporate that one as well. 

```suggestion
When this .NET Blazor Server starts and before listening for any HTTP requests, it bootstraps the application using a single-surface API provided as part of **Microsoft.Identity.Web** and **Microsoft.Idenitty.Web.UI**. The former is tying ASP.NET Core, its authentication middleware for sign-in, and the [Microsoft Authentication Library (MSAL) for .NET](https://github.com/azuread/microsoft-authentication-library-for-dotnet), while the latter adds UI components and controllers to facilitate user sign-in, sign-out, and other account experiences. All the details required for authentication are being gathered from a configuration section named `AzureAd` as well as others entries used during the call to a protected API, like the scopes. As for authorization, it is using the default policy options. Additionally, special services are injected [specifically for Blazor for re-signing, consent and conditional access purposes](https://github.com/AzureAD/microsoft-identity-web/wiki/Managing-incremental-consent-and-conditional-access#in-blazor-server). The recommended pattern to acquire tokens is first attempting to acquire them [non-interactevelly (or silent from cache), and then interactively](https://docs.microsoft.com/azure/active-directory/develop/msal-authentication-flows#interactive-and-non-interactive-authentication). Therefore, it is required to store the tokens for them to be cached. In this tutorial tokens are being stored in memory.
```